store/app/model.py

```diff
@@ -32,28 +33,57 @@ class RobolistBaseModel(BaseModel):
 class User(RobolistBaseModel):
     """Defines the user model for the API.
 
-    Users are defined by their email, username and password hash. This is the
-    simplest form of authentication, and is used for users who sign up with
-    their email and password.
+    Users are defined by their id and email (both unique).
+    Hashed password is set if user signs up with email and password, and is
+    left empty if the user signed up with Google or Github OAuth.
     """
 
-    email: str
-    permissions: set[UserPermission] | None = None
+    email: EmailStr
+    hashed_password: Optional[str] = None
+    permissions: Optional[Set[UserPermission]] = None
+    created_at: int
+    updated_at: int
+    email_verified_at: Optional[int] = None
+    github_id: Optional[str] = None
+    google_id: Optional[str] = None
```

Comment:
see above comment, this can just look like

```suggestion
    hashed_password: str | None = None
    permissions: set[UserPermission] | None = None
    created_at: int
    updated_at: int
    email_verified_at: int | None = None
    github_id: str | None = None
    google_id: str | None = None
```